tests/test_collection.py

```diff
@@ -9,101 +10,86 @@
-                {"type": "Point", "coordinates": (0, 3)},
-                {"type": "LineString", "coordinates": ((2, 0), (1, 0))}
-            ]}
-
-        # asShape
-        m = asShape(d)
-        self.assertEqual(m.geom_type, "GeometryCollection")
-        self.assertEqual(len(m), 2)
-        geom_types = [g.geom_type for g in m.geoms]
-        self.assertIn("Point", geom_types)
-        self.assertIn("LineString", geom_types)
-
-    def test_geointerface(self):
-        d = {"type": "GeometryCollection","geometries": [
-                {"type": "Point", "coordinates": (0, 3)},
-                {"type": "LineString", "coordinates": ((2, 0), (1, 0))}
-            ]}
-
-        # shape
-        m = shape(d)
-        self.assertEqual(m.geom_type, "GeometryCollection")
-        self.assertEqual(len(m), 2)
-        geom_types = [g.geom_type for g in m.geoms]
-        self.assertIn("Point", geom_types)
-        self.assertIn("LineString", geom_types)
-
-    @shapely20_deprecated
-    def test_empty_geointerface_adapter(self):
-        d = {"type": "GeometryCollection", "geometries": []}
-
-        # asShape
-        m = asShape(d)
-        self.assertEqual(m.geom_type, "GeometryCollection")
-        self.assertEqual(len(m), 0)
-        self.assertEqual(m.geoms, [])
-
-    def test_empty_geointerface(self):
-        d = {"type": "GeometryCollection", "geometries": []}
-
-        # shape
-        m = shape(d)
-        self.assertEqual(m.geom_type, "GeometryCollection")
-        self.assertEqual(len(m), 0)
-        self.assertEqual(m.geoms, [])
-
-    def test_empty_coordinates(self):
-
-        d = {"type": "GeometryCollection", "geometries": [
-            {"type": "Point", "coordinates": ()},
-            {"type": "LineString", "coordinates": (())}
-        ]}
-
-        # shape
-        m = shape(d)
-        self.assertEqual(m.geom_type, "GeometryCollection")
-        self.assertEqual(len(m), 0)
-        self.assertEqual(m.geoms, [])
-
-
-def test_geometrycollection_adapter_deprecated():
-    d = {"type": "GeometryCollection","geometries": [
-            {"type": "Point", "coordinates": (0, 3)},
-            {"type": "LineString", "coordinates": ((2, 0), (1, 0))}
+@pytest.fixture()
+def geometrycollection_geojson():
+    return {"type": "GeometryCollection", "geometries": [
+        {"type": "Point", "coordinates": (0, 3, 0)},
+        {"type": "LineString", "coordinates": ((2, 0), (1, 0))}
     ]}
+
+
+@pytest.mark.parametrize('geom', [
+    GeometryCollection(),
+    shape({"type": "GeometryCollection", "geometries": []}),
+    shape({"type": "GeometryCollection", "geometries": [
+        {"type": "Point", "coordinates": ()},
+        {"type": "LineString", "coordinates": (())}
+    ]}),
+    wkt.loads('GEOMETRYCOLLECTION EMPTY'),
+])
+def test_empty(geom):
+    assert geom.type == "GeometryCollection"
+    assert geom.type == geom.geom_type
+    assert len(geom) == 0
+    assert geom.geoms == []
```

Comment:
When #956 is addressed, the above two assertions will fail for the multi-part empty geometry. We'll get to that later however...

Suggest adding `assert geom.is_empty` here.